### DISCUSSION TITLE
keyball39 特定のキーの導通確認ができない。OLEDが表示されない。

### DISCUSSION THREAD

**Discussion by 529** [Yowkees/keyball#394] [Q&A] [answered]
お世話になります。確認ポイント改善点をご助言いただけますと幸いです。

現在 Build Guide for Keyball39 に従い組み立てを進めております。

機種：Keyball39 ver2
状況 : 
ステップ [3-7．ProMicroの書き込みと動作確認](https://github.com/Yowkees/keyball/blob/main/keyball39/doc/rev1/buildguide_jp.md#3-7promicro%E3%81%AE%E6%9B%B8%E3%81%8D%E8%BE%BC%E3%81%BF%E3%81%A8%E5%8B%95%E4%BD%9C%E7%A2%BA%E8%AA%8D) にて、
症状1 :
キースイッチのパッドへテスターをあてショートさせ直流電流を測定し反応を確認しておりますが、特定のキーのみ確認ができません。 (添付写真箇所)
症状2 :
両方のボードともUSB-C 接続し導通確認している際に、OLEDモジュールには何も表示されないです。
確認したこと：
ProMicro USB-C へ接続し片方づつ下記 2 点を確認しました。
動作しないキーのキースイッチのパッドにテスターを当てて、通電がないことを確認
その他の動作するキーのキースイッチのパッドにテスターをあてて、通電があることを確認

どうぞよろしくお願いいたします。

![image](https://github.com/Yowkees/keyball/assets/4526291/9e9faf2b-85f5-419f-84bc-04c38bab221d)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

順番に改善していきたいと思います。
まず、縦一列すべて入力できないキーについては、ProMicro裏の半田付けに問題がある場合が多いです。
ProMicro裏にはスルーホールがジグザグに配置されていて隣との距離が近いのでショートしやすいです。
隣とショートしているものが無いか順にご確認いただき、ショートしている部分には新しい半田を追加して形を丸型に整えると修正できます。

お手数ですがご確認お願いいたします。

**Reply by 529** [reply]
お世話になります。
ご回答ありがとうございます。

その後左手側 ProMicro に関しては、ピンソケットの状態が良くないと判断し、コンスルーにて代用し作成をすめました。
現在の状況といたしましては下記 3 点の症状となっております。

1. 左手写真赤枠のキーが反応しない
2. 左右両方の OLED が点灯しない
3. トラックボールの反応が不安定 (反応しない、もしくは反応しても軌道が急に飛んだりする)

お手数をおかけいたしますが、症状の改善に向けたアドバイスのほどどうぞよろしくお願いいたします。

![image](https://github.com/Yowkees/keyball/assets/4526291/b2868f07-50e8-43eb-9a0e-20396ccf422d)

![image](https://github.com/Yowkees/keyball/assets/4526291/0691c174-989a-47c9-ad3b-ee33e0e9a488)

**Reply by 529** [reply]
1. 左手写真赤枠のキーが反応しない
に関しましては、PCBソケット 並びに ダイオードのハンダを見直すことで解消いたしました。

残り下記 2 点について過去の事例などを参考に対処を試みてみます。
2. 左右両方の OLED が点灯しない
3. トラックボールの反応が不安定 (反応しない、もしくは反応しても軌道が急に飛んだりする)

もし気になる点などございましたらアドバイスいただけますと幸いです。

どうぞよろしくお願いいたします。

**Comment by 529** [comment]
ご確認ご返信ありがとうございます。

ProMicro裏のハンダを見直して、再度同じ確認を行いましたが状況は変わりませんでした。

隣とのショート確認 = テスターを利用した導通確認(電源はつながずに端子を順番に接続して数字の変化の有無を確認)したところ四角で囲んだところにおいて数字の変化はありますが、目視する限りではショートはしていないのではと考えております。
数字に変化があることは導通があるが、回路がつながっているからという理解です。私回路初心者であり深く理解ができておらず理解に不足がある可能性が大いにあります。
隣とのショートの確認方法に関して上記認識誤りなどありましたらご指摘いただけますと助かります。一通り過去のお問い合わせや、外部記事などを参照しておりますが、もし適切な記事などがありましたら共有いただけると助かります。

また、今回の組み立ての過程において、ProMicro用ピンソケット、trrsソケット、タクトスイッチ、4連ピンソケットをいずれも一度反対の面につけてしまい、はんだ吸い取り機を利用し外し、現在の状態に至っております。
作業中の目視の限りにおいては、破損などは見つけることはできていないですが、promicro用ピンソケットの一部ピンが抜けて差し直しをしている箇所もあります。

この場合、ピンソケットに問題がある可能性なども考えられるでしょうか。

自分の作業ミスでのリカバリに対するお問い合わせとなり恐縮ですがご助言いただけますと助かります。

![image](https://github.com/Yowkees/keyball/assets/4526291/e2a59305-e079-49a9-a9e2-239f24db4ffe)

**Comment by Yowkees** [comment, marked as answer]
ご対応ありがとうございます。

ProMicro裏にはショートしていそうな個所は無さそうということで了解しました。

うーんご指摘いただいた通り、ピンソケットを一度付け直したときに破損してしまった可能性が高そうです。ピンソケットのピンが一度脱落して付け直すというのは、かなりの確率で導通不要となります。

白銀ラボウェブショップよりご連絡いただきましたら、部品単位での販売も可能ですので、新しい部品や基板に交換を試みていただければ幸いです。

また、トラックボール基板の半田付け、7Pコンスルーの奥までしっかり基板が刺さっているかもトラックボール動作に影響しますので、一度ご確認いただければ幸いです。

以上、お手数ですがご検討お願いいたします。

**Comment by 529** [comment]
頂戴しましたアドバイスをもとに最終的には下記3点の問題に収束し、それぞれ解決いたしました。

その後左手側 ProMicro に関しては、ピンソケットの状態が良くないと判断し、コンスルーにて代用し作成をすめました。 その後の症状として下記3点ございましたがそれぞれ下記対処を実施し解決いたしました。

1.左手写真赤枠のキーが反応しない PCBソケット 並びに ダイオードのハンダを見直すことで解消いたしました。 2. 左右両方の OLED が点灯しない はんだの見直しで解消いたしました。
3. トラックボールの反応が不安定 (反応しない、もしくは反応しても軌道が急に飛んだりする)
はんだの見直しで解消いたしました

keyball 生活を楽しんでいこうと思います。
この度は貴重なアドバイスありがとうございました。
